ansible/roles/set-repositories/tasks/pre_checks_rhn.yml

```diff
@@ -35,3 +35,34 @@
         - rhsm_pool_ids is defined
         - rhel_subscription_user is not defined
         - rhel_subscription_pass is not defined
+
+- name: Ensure rhel_repo dictionary is exclusively defined
+  when:
+    - rhel_repos is defined
+  assert:
+    that: "{{ check.that }}"
+    fail_msg: "{{ check.msg }}"
+    quiet: true
+  loop_control:
+    loop_var: check
+    label: "{{ check.msg }}"
+  loop:
+    - msg: Ensure no exclusive variables are set.
+      that:
+        - rhel_repos_el7 is not defined
+        - rhel_repos_el8 is not defined
+
+- name: Ensure el7 and el8 dictionaries are exclusively defined
+  when:
+    - rhel_repos_el7 is defined or rhel_repos_el8 is defined
+  assert:
+    that: "{{ check.that }}"
+    fail_msg: "{{ check.msg }}"
+    quiet: true
+  loop_control:
+    loop_var: check
+    label: "{{ check.msg }}"
+  loop:
+    - msg: Ensure no exclusive variables are set.
+      that:
+        - rhel_repos is not defined
```

Comment:
What's going on with these loops with only one loop item? Why not combine all these `assert` tasks with loops with one item into a single assertion?